polynote-kernel/src/main/scala/polynote/kernel/dependency/CoursierFetcher.scala

```diff
@@ -0,0 +1,206 @@
+package polynote.kernel.dependency
+
+import java.io._
+import java.util.concurrent.{ConcurrentHashMap, Executors}
+import java.net.URI
+import java.nio.file.{Path, Paths}
+import java.util.concurrent.atomic.AtomicInteger
+
+import cats.data.{Validated, ValidatedNel}
+import cats.effect.{ContextShift, IO}
+import cats.implicits._
+import coursier.cache.{Cache, CacheLogger, FileCache}
+import coursier.ivy.IvyRepository
+import coursier.core._
+import coursier.error.ResolutionError
+import coursier.interop.cats._
+import coursier.params.ResolutionParams
+import coursier.{Artifacts, Attributes, Dependency, MavenRepository, Module, ModuleName, Organization, Repository, Resolution, Resolve}
+import polynote.config.{DependencyConfigs, RepositoryConfig, ivy, maven}
+import polynote.kernel.util.Publish
+import polynote.kernel._
+import polynote.messages.TinyString
+
+import scala.concurrent.ExecutionContext
+
+
+// Fetches only Scala dependencies
+class CoursierFetcher extends URLDependencyFetcher {
```

Comment:
It's not obvious in the diff because I moved it to a new file, but I was able to eliminate a lot of code from here thanks to the new Coursier APIs.